binr/r2r/load.c

```diff
@@ -198,17 +198,43 @@ R_API RPVector *r2r_load_cmd_test_file(const char *file) {
 			if (!strcmp (val, "1")) { \
 				test->field.value = true; \
 			} else if (!strcmp (val, "0")) { \
-                test->field.value = false; \
+				test->field.value = false; \
 			} else { \
 				eprintf (LINEFMT "Error: Invalid value \"%s\" for boolean key \"%s\", only \"1\" or \"0\" allowed.\n", file, linenum, val, key); \
 				goto fail; \
-            } \
+			} \
+			continue; \
+		}
+
+#define DO_KEY_NUM(key, field) \
+		if (strcmp (line, key) == 0) { \
+			if (test->field.value) { \
+				eprintf (LINEFMT "Warning: Duplicate key \"%s\"\n", file, linenum, key); \
+			} \
+			test->field.set = true; \
+			/* Strip comment */ \
+			char *cmt = strchr (val, '#'); \
+			if (cmt) { \
+				*cmt = '\0'; \
+				cmt--; \
+				while (cmt > val && *cmt == ' ') { \
+					*cmt = '\0'; \
+					cmt--; \
+				} \
+			} \
+			char *endval; \
+			test->field.value = strtol (val, &endval, 0); \
+			if (!endval || *endval) { \
```

Comment:
This condition is a bit strange ... https://linux.die.net/man/3/strtol says:

"If endptr is not NULL, strtol() stores the address of the first invalid character in *endptr. If there were no digits at all, strtol() stores the original value of nptr in *endptr (and returns 0). In particular, if *nptr is not '\0' but **endptr is '\0' on return, the entire string is valid."

I think `endval` here is pointing to the newline at the end.

Reply:
Oh sorry misread the manual -- the first condition was just to check whether `endval` is valid. My bad.